arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+          ? string.replace(reRegExpChar, '\\$&')
+          : string;
+      };
+    }());
+
+    var getKeysRegExp = function (map) {
+      var keys = Object.keys(map).sort(function (a, b) {
+        return b.length - a.length;
+      });
+
+      return new RegExp(keys.map(escapeRegExp).join('|'), 'gi');
+    };
+
+    var identifiersMap = MathJax.Hub.config.Arabic.identifiersMap;
+    var identifiersKeysRegExp = getKeysRegExp(identifiersMap);
+
+    var operatorsMap = MathJax.Hub.config.Arabic.operatorsMap;
+    var operatorsKeysRegExp = getKeysRegExp(operatorsMap);
+
+
+    TEX.Definitions.Add({
+      macros: {
+        'ar': 'HandleArabic',
+        'alwaysar': 'MarkAsArabic',
+        'fliph': 'HandleFlipHorizontal',
+        'transx': 'TranslateTeX',
+        'transt': 'TranslateText',
+        'transs': 'TranslateSymbols'
+      }
+    });
+
+
+    TEX.Definitions.Add({
+      macros: function () {
+        var definitions = {};
+
+        Object.keys(dict).forEach(function (key) {
+          var texCommand = dict[key][0];
+          definitions[texCommand] = key;
+        });
+
+        return definitions;
+      }()
+    });
+
+
+    TEX.Parse.Augment(function () {
+      var parsers = {};
+
+      Object.keys(dict).forEach(function (key) {
+        parsers[key] = dict[key][1]; // Parser function
+      });
+
+      return parsers;
+    }());
+
+
+    TEX.Parse.Augment({
+      flipHorizontal: function (token) {
+        token.arabicFlipH = !token.arabicFlipH;
+          // Invert the value, because flipping twice means, it is not flipped
+        return token;
+      },
+      arabicNumber: function (token) {
+        var numbersMap = MathJax.Hub.config.Arabic.numbersMap;
+        var text = token.data[0].data[0];
+        var mapped = text.replace(englishNumbersRegExp, function (m) {
+          return numbersMap[m];
+        });
+
+        if (mapped !== text) {
+          token.data[0].data[0] = mapped;
+          token.arabicFontLang = 'ar';
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicIdentifier: function (token) {
+        var text = token.data[0].data[0];
+
+        if ('chars' === token.data[0].type) {
+          // English Symbols like X and Y
+          var mapped = text.replace(identifiersKeysRegExp, function (m) {
+            return identifiersMap[m.toLowerCase()];
+          });
+
+          if (mapped !== text) {
+            token.data[0].data[0] = mapped;
+            token.arabicFontLang = 'ar';
+          }
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicOperator: function (token) {
+        var text = token.data[0].data[0];
+        var mapped = text.replace(operatorsKeysRegExp, function (m) {
+          return operatorsMap[m];
+        });
+
+        if (mapped !== text) {
+          token = this.flipHorizontal(token);
+          token.arabicFontLang = 'ar';
+          token.data[0].data[0] = mapped;
+        }
```

Comment:
Is there a reason that the `arabicOperator()` works differently from the other two `arabicNumber()` and `arabicIdentifier()`?  That is, here the `flipHorizontal()` call is only performed if `mapped` is not equal to `text`, but in the other two, it is always done.